### ISSUE TITLE
How to setup sync on a second browser without destroying the bookmarks?

### ISSUE THREAD

**Issue by CajusH** [marcelklehr/floccus#111]
### Software versions

* Browser: Firefox 60.0.1 64 bit on openSUSE 42.3 and openSUSE 15.0
* Nextcloud: 13.0.1 (still using sqlite DB, which is the Installation default)
* Nextcloud Bookmarks app: 0.11.0
* Floccus: 2.2.5
**Please read until the end, some issues solved themselves, but might be worth a note in the installation docu or some prevention in floccus...**

### Steps to reproduce

1. Empty DB on nextcloud, initial sync with firefox from computer no. 1
2. nextcloud DB has 457 bookmark entries which looks OK
3. Deleted all bookmarks on the second computers firefox and installed Floccus
4. Setup Floccus and "Sync Now". Nothing happend, no bookmarks appear.
5. I found the bug #83 , so I exported all bookmarks from computer no.1 and imported it into computer no.2.
6. I clicked "Sync Now" on computer no.2 again. Floccus aborted with "Cannot delete bookmark entry in the DB, index 507" (Maybe the text was a little different, I did not write it down at the moment it appeared) Index 507? There is no such index in the nextcloud DB?
7. I went back to the computer no.1 and clicked "Sync Now"
Half of the bockmarks disappeared one after each other... :-(

### Expected outcome

It should be possible do do an "initial sync" on a second browser, with the option to remove all the old bookmarks. This should also work on a fresh browse, with no bookmarks existing at all.
...

### Actual outcome

I had to restore the bookmarks on both browsers (via import of my saved bookmarks file)
On computer no.1, I clicked "Sync Now" again after the restore and it seem to run forever.
The dialog says "Syncing..." and stays like this (since 20 minutes now), even after restart of firefox.
I deleted the floccus account on computer no.1 and created a new one (with the same user and options). "Sync Now" worked again on this computer (no destroying of bookmarks)
Back to computer no.2: I noticed a message "Reconnect to Sync" on this computer. This Firefox had been connected to an old local firefox sync server, which isn't working with FF 60 any more. I deleted the complete .mozilla folder and setup firefox from scratch. Starting with all bookmarks deleted (even those that are created by firefox on empty preferences) It synced with nextcloud without an error!
All the bookmarks appeared EXCEPT those in the "Lesezeichen-Symbolleiste". So I copied the complete "Lesezeichen-Symbolleiste"-folder into my "Lesezeichen-Menu"-folder and did a Sync.
This new folder did NOT appear on computer no.2 after I did a sync there!??
Then I added one item to the "Lesezeichen-Symbolleiste" on computer no. 2 - Sync - and this one appeared on computer no. 1 after resync.
Delete the new item on computer no. 1 also deletes the item on no. 2 after sync, but the original items from the "Lesezeichen-Symbolleiste" did never showed up, neither as "Lesezeichen-Symbolleiste" or as folder inside my "Lesezeichen-Menu"-folder.
OOPS: The name  "Lesezeichen-Symbolleiste" seems to be special to firefox. You cannot edit the name of a folder with the name "Lesezeichen-Symbolleiste" **at any place inside your bookmarks**
I just moved the bookmarks out of my folder "Lesezeichen-Menu"-"Lesezeichen-Symbolleiste" (which was the copy from the root -"Lesezeichen-Symbolleiste")  **and all bookmarks from the root -"Lesezeichen-Symbolleiste" disappeaed as well**
Then I copied the bookmarks back to the root -"Lesezeichen-Symbolleiste" and deleted my folder "Lesezeichen-Menu"-"Lesezeichen-Symbolleiste". The bookmarks are now at two places: the root -"Lesezeichen-Symbolleiste" and in a folder "Lesezeichen-Menu"-"dummy".
Now I cannot sync any more because floccus complains _same URL is bookmared twice_
...

**Comment by CajusH** [comment]
In the mean time both browsers seem to work as expected.
I created two accounts on each browser: one for "Lesezeichen-Symbolleiste" and and one for "Lesezeichen-Menu". Both accounts write to a path with the same name on nextcloud. With this configuration you can sync both.
I had one more issue with bookmarks getting lost on sync, but I couldn't reproduce it.
When the sync on computer no. 1 runs and/or get stuck, a sync on computer no.2 seem to trash half of the bookmarks.

**Comment by CajusH** [comment]
I'm sorry, I wrote the bug report during the testing and trying. The result is a spaghetti-like report :-) 
I saw you have released 3 more versions in the meantime.
With 2.2.7 I did not have problems with disappearing bookmarks again.
Sync from my 2 computers seem to run now without getting stuck.
Currently I only have 2 minor open issues:
1.) bookmarks deleted from one computer don't get deleted on the second computer and may re-appear on the first computer after syncing the second computer
2.) The order of the bookmarks gets mixed sometimes, which is not nice.
